src/User/Keycloak/CachedUserIdentityResolver.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\User\Keycloak;
+
+use CultuurNet\UDB3\Model\ValueObject\Web\EmailAddress;
+use CultuurNet\UDB3\User\UserIdentityDetails;
+use CultuurNet\UDB3\User\UserIdentityResolver;
+use Symfony\Contracts\Cache\CacheInterface;
+
+final class CachedUserIdentityResolver implements UserIdentityResolver
+{
+    private UserIdentityResolver $userIdentityResolver;
+
+    private CacheInterface $cache;
+
+    public function __construct(
+        UserIdentityResolver $userIdentityResolver,
+        CacheInterface $cache
+    ) {
+        $this->userIdentityResolver = $userIdentityResolver;
+        $this->cache =$cache;
+    }
+    public function getUserById(string $userId): ?UserIdentityDetails
+    {
+        return $this->cache->get(
+            $this->createCacheKey($userId, 'user_id'),
+            function () use ($userId) {
+                return $this->userIdentityResolver->getUserById($userId);
+            }
+        );
+    }
+
+    public function getUserByEmail(EmailAddress $email): ?UserIdentityDetails
+    {
+        return $this->cache->get(
+            $this->createCacheKey($email->toString(), 'email'),
+            function () use ($email) {
+                return $this->userIdentityResolver->getUserByEmail($email);
```

Comment:
I wonder how this PHP object gets stored inside the cache? Maybe we need to encode it as an associative array?